search/mongo_search_test.go

```diff
@@ -455,17 +455,23 @@ func (m *MongoSearchSuite) TestConditionReferenceQueryObjectByPatientGender(c *C
 }
 
 func (m *MongoSearchSuite) TestConditionReferenceQueryByPatientGender(c *C) {
+	var result interface{}
+
 	q := Query{"Condition", "patient.gender=male"}
-	mq := m.MongoSearcher.CreateQuery(q)
-	num, err := mq.Count()
+	result = models.NewSlicePlusForResourceName(q.Resource, 0, 0)
```

Comment:
Actually, if there is no _include or _revinclude, then I think the search should just return the normal structs, not the sliceplus structs.  More on that a bit further done in the code review.

Also, since you know you're querying on Condition, don't use `models.NewSliceForResourceName` -- just use `[]*models.Condition`.